### PR TITLE
Issue#71 poc

### PR DESCRIPTION
**Pull request by OlTym**
dev
## ZenHub

* [Main ZenHub ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: home-application/pom.xml [hunk 1]
```diff
@@ -13,6 +13,11 @@
 
     <artifactId>home-application</artifactId>
     <dependencies>
+        <dependency>
+            <groupId>io.github.perplexhub</groupId>
+            <artifactId>rsql-jpa-spring-boot-starter</artifactId>
+            <version>5.0.6</version>
```
Comment (Romanov-Niko) [inline home-application/pom.xml]: Move versions to property section please

File: home-data/pom.xml [hunk 2]
```diff
@@ -19,6 +19,14 @@
     </properties>
 
     <dependencies>
+
+        <dependency>
+            <groupId>io.github.perplexhub</groupId>
+            <artifactId>rsql-jpa-spring-boot-starter</artifactId>
+            <version>5.0.6</version>
```
Comment (Romanov-Niko) [inline home-data/pom.xml]: Move versions to property section please

File: home-application/src/main/resources/config/application.properties [hunk 3]
```diff
@@ -1,3 +1,4 @@
+#spring.profiles.active=dev
```
Comment (Romanov-Niko) [inline home-application/src/main/resources/config/application.properties]: We are going to delete dataloader and we don't need this profile

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/config/DataLoader.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                .password(passwordEncoder.encode("password"))
+                .expired(false)
+                .locked(false)
+                .credentialsExpired(false)
+                .enabled(true)
+                .roles(new HashSet<>() {{add(role);}})
+                .build());
+    }
+
+}
```
Comment (Romanov-Niko) [inline home-application/src/main/java/com/softserveinc/ita/homeproject/application/config/DataLoader.java]: Data loader is going to be deleted

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/UserApiServiceImpl.java [hunk 5]
```diff
@@ -77,10 +81,17 @@ public Response getUser(Long id) {
      * @param pageSize is amount of the returned elements
      * @return Response to generated controller
      */
-    @PreAuthorize(GET_ALL_USERS_PERMISSION)
```
Comment (Romanov-Niko) [inline home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/UserApiServiceImpl.java:80]: Did we really need to remove this permission?

**Review by Romanov-Niko — CHANGES_REQUESTED** [review]
There are small remarks about refactoring

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/config/DataLoader.java (hunk 4 above)
Comment (OlTym) [reply home-application/src/main/java/com/softserveinc/ita/homeproject/application/config/DataLoader.java]: agree I use dataLoader to insert users into local date fastly.

File: home-application/src/main/resources/config/application.properties (hunk 3 above)
Comment (OlTym) [reply home-application/src/main/resources/config/application.properties]: Agree But now dataloader is in the project becouse I did not delete this commented code

File: home-data/pom.xml (hunk 2 above)
Comment (OlTym) [reply home-data/pom.xml]: Agree

File: home-application/pom.xml (hunk 1 above)
Comment (OlTym) [reply home-application/pom.xml]: Agree

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/UserApiServiceImpl.java (hunk 5 above)
Comment (OlTym) [reply home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/UserApiServiceImpl.java:80]: Agree this permition is nessesary

**Review by Abhai2016 — APPROVED** [review]

**Review by alexorlenko — APPROVED** [review]

**Review by Dandria1802 — APPROVED** [review]

**Review by PoLytvy — APPROVED** [review]

**Review by Romanov-Niko — APPROVED** [review]

**Pull request closed without merging**